components/camel-hazelcast/src/main/java/org/apache/camel/processor/aggregate/hazelcast/HazelcastAggregationRepository.java

```diff
@@ -290,6 +290,15 @@ public Exchange get(CamelContext camelContext, String key) {
         return unmarshallExchange(camelContext, cache.get(key));
     }
     
+    /**
+     * Checks if the key in question is in the repository.
+     * 
+     * @param key Object - key in question
+     */
+    public boolean containsKey(Object key){
+        return cache.containsKey(key);
```

Comment:
you are looking only in the cache, you need to see if this value can be loaded also.
There is also a persistedCache, is persistedCache a subset of the ache or do we need to look into both?